packages/navigator/src/browser/open-editors-widget/navigator-open-editors-tree-model.ts

```diff
@@ -84,7 +88,15 @@ export class OpenEditorsModel extends FileTreeModel {
                 this.selectNode(nodeToSelect);
             }
         }));
-        this.toDispose.push(this.applicationShell.onDidAddWidget(() => this.updateOpenWidgets()));
+        this.toDispose.push(this.applicationShell.onDidAddWidget(async () => {
+            await this.updateOpenWidgets();
+            const existingWidgetIds = this.editorWidgets.map(widget => widget.id);
```

Comment:
Maybe make this a `Set` so that it's easier to check for membership?

Reply:
Nice idea